src/client/dfs/obj.c

```diff
@@ -510,7 +511,8 @@ dfs_dup(dfs_t *dfs, dfs_obj_t *obj, int flags, dfs_obj_t **_new_obj)
 		D_GOTO(err, rc = EINVAL);
 	}
 
-	strncpy(new_obj->name, obj->name, DFS_MAX_NAME + 1);
```

Comment:
Again this is fine too if the `obj->name` is already valid.

I think a serious issue is about the consistency of using `DFS_MAX_NAME`. Sometimes the name is defined as `name[DFS_MAX_NAME + 1]` but sometimes it's `name[DFS_MAX_NAME]`. I would prefer to use the latter and  then redefining maximum object name as `DFS_MAX_NAME` including the ending zero.

Reply:
Again, this is the result of #15105 + see my comment above: https://github.com/daos-stack/daos/pull/15920#issuecomment-2664838870